Submission Contact Fragment & StepView Custom View

Fabian-K opened the pull request:
## Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] If this PR comes from a fork, please [Allow edits from maintainers](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
* [x] Set a speaking title. Format: {task_name} (closes #{issue_number}). For example: Use logger (closes #41)
* [x] [Link your Pull Request to an issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) (if applicable)
* [x] Create Work In Progress [WIP] pull requests only if you need clarification or an explicit review before you can continue your work item.
* [x] Make sure that your PR is not introducing _unncessary_ reformatting (e.g., introduced by on-save hooks in your IDE)
* [x] Check our [Contribution Guidelines](https://github.com/corona-warn-app/cwa-app-android/blob/master/CONTRIBUTING.md)

## Description
This PR adds the Submission Contact Fragment which allows to call for a teletan. 

![image](https://user-images.githubusercontent.com/862577/83517177-b680b180-a4d8-11ea-95f0-77cab743fb25.png)

It also includes a custom view StepEntry that can be reused to display fragments like the following: 
![image](https://user-images.githubusercontent.com/862577/83517284-e92aaa00-a4d8-11ea-8bac-debe5935c875.png)
The use of a custom view compared to `include_test_result_step.xml` is more flexible with respect to the body and enables composability. Also it renders correctly in the layout preview. If accepted, I plan to replace include_test_result_step with SimpleStepEntry in a following PR.

Looking forward to your feedback!

Thanks, Fabian

Fabian-K committed:
- submission-contact fragment added
- StepEntry custom view added to unify various step entry views
- SimpleStepEntry as a simple implementation of a StepEntry with title and text added

harambasicluka committed:
Merge branch 'dev' into feature/submission-contact

harambasicluka commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/SubmissionContactFragment.kt:
```diff
@@ -0,0 +1,53 @@
+        }
+    }
+
+    private fun dial() = context?.let {
```
we already have a CallHelper in place, please use this one.

harambasicluka commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/view/StepEntry.kt:
```diff
@@ -0,0 +1,51 @@
+
+        context.withStyledAttributes(attrs, R.styleable.StepEntry) {
+            val icon = getDrawable(R.styleable.StepEntry_step_entry_icon)
+            findViewById<ImageView>(R.id.step_entry_imageview_icon).setImageDrawable(icon)
```
please use `kotlinx.android.synthetic...` to access the view

harambasicluka commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/view/StepEntry.kt:
```diff
@@ -0,0 +1,51 @@
+            findViewById<ImageView>(R.id.step_entry_imageview_icon).setImageDrawable(icon)
+
+            val isFinal = getBoolean(R.styleable.StepEntry_step_entry_final, false)
+            findViewById<View>(R.id.step_entry_view_line).visibility = if (isFinal) {
```
please use `kotlinx.android.synthetic...` to access the view

harambasicluka commented on Corona-Warn-App/src/main/res/layout/include_step_entry_simple_body.xml:
```diff
@@ -0,0 +1,28 @@
+    android:layout_height="wrap_content">
+
+    <TextView
+        android:id="@+id/simple_step_entry_textview_title"
```
we don't use textview in the id, the type of the element should be clear through `_title`

harambasicluka commented on Corona-Warn-App/src/main/res/layout/include_step_entry_simple_body.xml:
```diff
@@ -0,0 +1,28 @@
+        tools:text="Title" />
+
+    <TextView
+        android:id="@+id/simple_step_entry_textview_body"
```
we don't use textview in the id, the type of the element should be clear through `_body`

harambasicluka commented on Corona-Warn-App/src/main/res/layout/include_submission_contact.xml:
```diff
@@ -0,0 +1,109 @@
+            android:layout_height="wrap_content">
+
+            <ImageView
+                android:id="@+id/submission_contact_imageview_illustration"
```
we don't use imageview in the id, the type of the element should be clear through `_illustration`

harambasicluka commented on Corona-Warn-App/src/main/res/layout/include_submission_contact.xml:
```diff
@@ -0,0 +1,109 @@
+                app:layout_constraintTop_toTopOf="parent" />
+
+            <TextView
+                android:id="@+id/submission_contact_textview_body"
```
we don't use textview in the id, the type of the element should be clear through `_body`

harambasicluka commented on Corona-Warn-App/src/main/res/layout/include_submission_contact.xml:
```diff
@@ -0,0 +1,109 @@
+                app:layout_constraintTop_toBottomOf="@id/submission_contact_imageview_illustration" />
+
+            <TextView
+                android:id="@+id/submission_contact_textview_headline"
```
we don't use textview in the id, the type of the element should be clear through `_headline`

harambasicluka commented on Corona-Warn-App/src/main/res/layout/view_step_entry.xml:
```diff
@@ -0,0 +1,49 @@
+            app:layout_constraintTop_toTopOf="parent" />
+
+        <FrameLayout
+            android:id="@+id/step_entry_framelayout_body"
```
looks like an unused FrameLayout

Fabian-K replied:
This is the insertion point for the children of the StepEntry:

```
<de.rki.coronawarnapp.ui.view.StepEntry>
  <TextView
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:text="@string/submission_contact_step_2_body" />
</de.rki.coronawarnapp.ui.view.StepEntry>
```

harambasicluka commented on Corona-Warn-App/src/main/res/layout/view_step_entry.xml:
```diff
@@ -0,0 +1,49 @@
+        android:layout_height="wrap_content">
+
+        <ImageView
+            android:id="@+id/step_entry_imageview_icon"
```
remove imageview from id

harambasicluka commented on Corona-Warn-App/src/main/res/layout/include_submission_contact.xml:
```diff
@@ -0,0 +1,109 @@
+                        app:layout_constraintTop_toTopOf="parent" />
+
+                    <TextView
+                        android:id="@+id/submission_contact_textview_step_1_number"
```
remove textview

harambasicluka commented on Corona-Warn-App/src/main/res/layout/include_submission_contact.xml:
```diff
@@ -0,0 +1,109 @@
+                    android:layout_height="wrap_content">
+
+                    <TextView
+                        android:id="@+id/submission_contact_textview_step_1_body"
```
we don't use textview in the id, the type of the element should be clear through `_title`

harambasicluka reviewed (changes requested):
The ID misunderstandings are due to a non-unique [naming convention](https://github.com/corona-warn-app/cwa-app-android/wiki/3-UI#view-ids).

Fabian-K committed:
- apply naming conventions for ids
- use CallHelper to trigger a call
- access views using kotlinx.android.synthetic

Fabian-K commented:
🙏 for the feedback! I applied the changes and added one comment regarding the FrameLayout.

harambasicluka committed:
Merge branch 'dev' into feature/submission-contact
